### DISCUSSION TITLE
Dúvida Modelando domínios ricos

### DISCUSSION THREAD

**Discussion by matheusvarela** [balta-io/forum#34] [Tira Dúvidas] [answered]
Boa noite Balta

Fiz o curso modelando dominios ricos e fiquei com algumas dúvidas 

Vi que foi criado apenas uma interface de repositorio para persistir a entidade Student (e entidades relacionadas),mas as entidades relacionadas não terão seus repositorios? ou essas entidades "relacionadas" fazem parte de um agregado Student ?

É correto, repositorios acessar outros repositórios?

No caso da Subscription, dentro dela possui uma lista de entidades Payment, para persistir essa lista eu teria que ter ifs que verificam se os payments são paypal ou boleto ou cartão?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @matheusvarela, como vai?

Se não me engano, neste caso estas entidades dazem parte do Student que atua como um agregado raiz. Desta forma não vejo problemas em ter um repo apenas... porém se precisar de mais métodos para interagir com outras entidades, é bom separar.

Então não tem uma regra de ouro, vai da demanda... Começa com um repo e depois vai seraparando a medida que precisa.

Sobre ter IFs no repositório, eu acho ruim... cria métodos separados e deixa estes IFs nos Handlers.
